File: ETFEase (Android)/src/com/etfease/datasource/Trades.java
package com.etfease.datasource;

import java.util.ArrayList;
import java.util.Iterator;
import java.util.List;
public class Trades {
	private ETFFilterType etfFilterType;
	private float total;
	private List<Trade> trades = new  ArrayList <Trade> ();
	private boolean recentBuy;
	private boolean recentSell;
	private boolean buyNow;
	private boolean sellNow;
	private String lastTradeDate;
	
	

	public static final Filter<Trades> R3 = 
		new Filter<Trades>() {
		public boolean passes(Trades trades) {
			return trades.getEtfFilterType().name().equals("R3");
		}
	};	

	public static final Filter<Trades> RSI25 = 
		new Filter<Trades>() {
		public boolean passes(Trades trades) {
			return trades.getEtfFilterType().name().equals("RSI25");
		}
	};	
	
	public static final Filter<Trades> RSI75 = 
		new Filter<Trades>() {
		public boolean passes(Trades trades) {
			return trades.getEtfFilterType().name().equals("RSI75");
		}
	};	
	
	
	
	public void addTrade(float shares, float tradeAmount, float remainingAmount, float fee, String date){
		trades.add(new Trade(shares, tradeAmount, remainingAmount, fee, date));
	}
	
	public float getTotal(){
		return total;
	}
	
	public List <Trade> getTrades(){
		return trades;
	}




	/**
	 * @return the etfFilter
	 */
	public ETFFilterType getEtfFilterType() {
		return etfFilterType;
	}

	/**
	 * @param etfFilter the etfFilter to set
	 */
	public void setEtfFilterType(ETFFilterType etfFilterType) {
		this.etfFilterType = etfFilterType;
	}

	/**
	 * @param total the total to set
	 */
	public void setTotal(float total) {
		this.total = total;
	}

	public String toString(){
		StringBuffer sb = new StringBuffer();
		Iterator <Trade> trd= trades.iterator();
		while(trd.hasNext()){
			sb.append(trd.next().toString());
			sb.append("\n");
		}
		return sb.toString();
	}

	public boolean isRecentBuy() {
		return recentBuy;
	}

	public void setRecentBuy(boolean recentBuy) {
		this.recentBuy = recentBuy;
	}

	public boolean isRecentSell() {
		return recentSell;
	}

	public void setRecentSell(boolean recentSell) {
		this.recentSell = recentSell;
	}

	public boolean isBuyNow() {
		return buyNow;
	}

	public void setBuyNow(boolean buyNow) {
		this.buyNow = buyNow;
	}

	public boolean isSellNow() {
		return sellNow;
	}

	public void setSellNow(boolean sellNow) {
		this.sellNow = sellNow;
	}

	public String getLastTradeDate() {
		return lastTradeDate;
	}

	public void setLastTradeDate(String lastTradeDate) {
		this.lastTradeDate = lastTradeDate;
	}

}

File: ETFEase (iOS)/DataSource/Trades.m
//
//  Trades.m
//  Untitled
//
//  Created by Michael Lindeboom on 11/21/10.
//  Copyright 2010 __MyCompanyName__. All rights reserved.
//

#import "Trades.h"
#import "Trade.h"


@implementation Trades

@synthesize total;
@synthesize trades;
@synthesize etfFilterType;
@synthesize ETFFilterTypeAsString;
@synthesize isRecentBuy;
@synthesize isRecentSell;
@synthesize isBuyNow;
@synthesize isSellNow;


-(id)init{
    if (self = [super init])
    {
		self.ETFFilterTypeAsString = [NSArray arrayWithObjects:	@"R3", @"RSI25", @"RSI75", nil];
		NSMutableArray *mut =  [[NSMutableArray alloc] init]; 
		self.trades = mut;
		[mut release];
    }
    return self;
}

-(void) dealloc{
	[total release];
	[trades release];
	[etfFilterType release];
	[ETFFilterTypeAsString release];
	[super dealloc];
}


-(void) addTrade:(NSNumber *) inShares tradeAmount:(NSNumber *) inTradeAmount fee:(NSNumber *) inFee date:(NSString *) inDate remainingAmount:(NSNumber *)  inRemainingAmount {
	//create a trade object instance and add it to the trades list
	Trade* myTrade = [[Trade alloc] initWithTrade:inShares tradeAmount:inTradeAmount fee:inFee date:inDate remainingAmount:inRemainingAmount];
	[trades addObject:myTrade];
	[myTrade release]; 
}



-(BOOL) recentBuy{
	
	int last = [trades count];
	
	if (last==0) {
		return NO;
	} else last--;
	
	Trade *myTrade = [trades objectAtIndex:last];
	
	if ([[myTrade shares] intValue]>0) {
		return YES;
	} else {
		return NO;
	}
}


-(BOOL) recentSell{
	int last = [trades count];
	
	if (last==0) {
		return NO;
	} else last--;
	
	Trade *myTrade = [trades objectAtIndex:last];
	NSDateFormatter *dateFormatter = [[[NSDateFormatter alloc] init] autorelease];
	[dateFormatter setDateFormat:@"yyyy-MM-dd"];
	NSDate *myTradeDate = [dateFormatter dateFromString:[myTrade date]];
	NSDate *todaysDate = [NSDate date];
	NSTimeInterval secondsSinceTrade = [todaysDate timeIntervalSinceDate: myTradeDate];
	
	if ([[myTrade shares] intValue]<0 && secondsSinceTrade/86400<=7) return YES;
	else return NO;
}





-(BOOL) buyNow{
	
	int last = [trades count];
	
	if (last==0) {
		return NO;
	} else last--;
	
	Trade *myTrade = [trades objectAtIndex:last];
	NSDateFormatter *dateFormatter = [[[NSDateFormatter alloc] init] autorelease];
	[dateFormatter setDateFormat:@"yyyy-MM-dd"];
	NSDate *myTradeDate = [dateFormatter dateFromString:[myTrade date]];
	NSDate *todaysDate = [NSDate date];
	NSTimeInterval secondsSinceTrade = [todaysDate timeIntervalSinceDate: myTradeDate];
	
	if ([[myTrade shares] intValue]>0 && secondsSinceTrade/86400<=1) return YES;
	else return NO;
	
}




-(BOOL) sellNow{
	int last = [trades count];
	
	if (last==0) {
		return NO;
	} else last--;
	
	Trade *myTrade = [trades objectAtIndex:last];
	NSDateFormatter *dateFormatter = [[[NSDateFormatter alloc] init] autorelease];
	[dateFormatter setDateFormat:@"yyyy-MM-dd"];
	NSDate *myTradeDate = [dateFormatter dateFromString:[myTrade date]];
	NSDate *todaysDate = [NSDate date];
	NSTimeInterval secondsSinceTrade = [todaysDate timeIntervalSinceDate: myTradeDate];
	
	if ([[myTrade shares] intValue]<0 && secondsSinceTrade/86400<=1) return YES;
	else return NO;
	
}




@end
